src/components/Main.tsx

```diff
@@ -0,0 +1,687 @@
+        }}
+      />
+
+      <Stack.Screen name="fioAddressList" component={ifLoggedIn(FioAddressListScene)} />
+
+      <Stack.Screen
+        name="fioAddressRegister"
+        component={ifLoggedIn(FioAddressRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSelectWallet"
+        component={ifLoggedIn(FioAddressRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_fio_address_confirmation,
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainRegister"
+        component={ifLoggedIn(FioDomainRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainRegisterSelectWallet"
+        component={ifLoggedIn(FioDomainRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_register_fio_domain,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainConfirm"
+        // @ts-expect-error
+        component={ifLoggedIn(FioNameConfirmScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioNameConfirm"
+        component={ifLoggedIn(FioNameConfirmScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressDetails"
+        component={ifLoggedIn(FioAddressDetailsScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressDetails'> fromParams={params => params.fioAddressName} />
+        }}
+      />
+      <Stack.Screen
+        name="fioConnectToWalletsConfirm"
+        component={ifLoggedIn(FioConnectWalletConfirmScene)}
+        options={{
+          title: s.strings.title_fio_connect_to_wallet
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressSettings"
+        component={FioAddressSettingsScene}
+        options={{
+          title: s.strings.title_fio_address_settings
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSuccess"
+        component={ifLoggedIn(FioAddressRegisteredScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressRegisterSuccess'> fromParams={params => params.fioName} />,
+          headerLeft: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainSettings"
+        component={FioDomainSettingsScene}
+        options={{
+          title: s.strings.title_fio_domain_settings
+        }}
+      />
+
+      <Stack.Screen name="fioRequestList" component={ifLoggedIn(FioRequestListScene)} />
+      <Stack.Screen
+        name="fioRequestApproved"
+        // @ts-expect-error
+        component={ifLoggedIn(TransactionDetailsScene)}
+        listeners={{ focus: async () => requestPermission('contacts') }}
```

Comment:
Also this `async` should go away, and just do `.catch(showError)`.

Reply:
Ah.